lib/puppet/type/file.rb

```diff
@@ -393,6 +393,22 @@ def self.title_patterns
 
     self.warning "Checksum value is ignored unless content or source are specified" if self[:checksum_value] && !self[:content] && !self[:source]
 
+    if Puppet.features.microsoft_windows? && parameter(:source) &&
+      [:use, :use_when_creating].include?(self[:source_permissions])
+      err_msg = "Copying owner/mode/group from the source file on Windows" <<
+      " is not supported; use source_permissions => ignore."
+
+      if self[:owner] == nil || self[:owner] == :absent || self[:group] == nil ||
+        self[:group] == :absent || self[:mode] == nil || self[:mode] == :absent
```

Comment:
`puppet inspect` relies on the cached catalog, so I'm not sure how you would ever get a cached catalog that has `source_permissions => use` and mode/owner/group marked absent. I think we can skip checking `:absent`, and rely on the devfail.